src/networking.c

```diff
@@ -117,35 +117,19 @@ int authRequired(client *c) {
     return auth_required;
 }
 
-client *createClient(connection *conn) {
-    client *c = zmalloc(sizeof(client));
-
-    /* passing NULL as conn it is possible to create a non connected client.
-     * This is useful since all the commands needs to be executed
-     * in the context of a client. When commands are executed in other
-     * contexts (for instance a Lua script) we need a non connected client. */
-    if (conn) {
-        connEnableTcpNoDelay(conn);
-        if (server.tcpkeepalive)
-            connKeepAlive(conn,server.tcpkeepalive);
-        connSetReadHandler(conn, readQueryFromClient);
-        connSetPrivateData(conn, c);
-    }
-
+static void initClientVariables(client *c) {
     selectDb(c,0);
     uint64_t client_id;
     atomicGetIncr(server.next_client_id, client_id, 1);
     c->id = client_id;
     c->resp = 2;
-    c->conn = conn;
+    c->conn = NULL;
```

Comment:
maybe i don't understand the responsibility of initClientVariables, but i think it should probably not reset `conn` and `id`

Reply:
`initClientVariables()` is called just after `malloc(sizeof(client)` or when we want to reuse client object, it that case, we clear it as if it is a brand new allocation. Maybe change `clearClient()` name, my intention was `clearClient()` will destroy and create new `client` object without deallocating its memory. After `clearClient()` is called on a client object, it's a brand new client object.

Reply:
ok, so it assumes nothing there is being used (i.e. clientUnlink was called).
maybe rename to `initClientStruct`?
please try to define / describe / document these functions and their roles, maybe then it'll make more sense to me.

Reply:
p.s. when we first suggested that module.c and resetCommand can share code, i imagined a much smaller change.
i.e. extracting some code from resetCommand into a function to be used by both.
maybe we got carried away?

Reply:
After discussion with @yossigo , I tried to create a generic function to clear all kinds of clients, maybe I misunderstood. So, it can be used different places later if needed etc.